### PR TITLE
feat(RBAC): Authorize actions,resources mapping

### PR DESCRIPTION
**Pull request by moogacs**
### What's being changed:

this PR reflect the table below in order to make sure every request has the correct combination of (verb, resource)

| Domain     | Verb | Resource                               | Path                                                           |
|--------------|------------|---------------------------------------|----------------------------------------------------------------|
| Roles        | CRUD       |*                                       | roles/*                                                        |
| Roles        | R          | role name                             | roles/{role_name}                                              |
| Cluster      | CRUD       |*                                       | cluster/*                                                      |
| Collections  | C          | *                                      | collections/*                                                  |
| Collections  | R          | collection name                       | collections/{collection_name}                                   |
| Collections  | U          | collection name                       | collections/{collection_name}                                   |
| Collections  | D          | collection name                       | collections/{collection_name}                                   |
| Tenants      | C          | collection name                       | collections/{collection_name}/shards/*                          |
| Tenants      | R          | collection name, tenant name          | collections/{collection_name}/shards/{shard_name}               |
| Tenants      | U          | collection name, tenant name          | collections/{collection_name}/shards/{shard_name}               |
| Tenants      | D          | collection name, tenant name          | collections/{collection_name}/shards/{shard_name}               |
| Objects      | C          | collection name, tenant name          | collections/{collection_name}/shards/{shard_name}/objects/*     |
| Objects      | R          | collection name, tenant name, object id | collections/{collection_name}/shards/{shard_name}/objects/{object_id} |
| Objects      | U          | collection name, tenant name, object id | collections/{collection_name}/shards/{shard_name}/objects/{object_id} |
| Objects      | D          | collection name, tenant name, object id | collections/{collection_name}/shards/{shard_name}/objects/{object_id} |

### Review checklist

- [ ] Documentation has been updated, if necessary. Link to changed documentation:
- [ ] Chaos pipeline run or not necessary. Link to pipeline:
- [ ] All new code is covered by tests where it is reasonable.
- [ ] Performance tests have been run or not necessary.

### CODE REVIEW

File: usecases/auth/authorization/adminlist/authorizer.go [hunk 1]
```diff
@@ -38,7 +38,7 @@ func New(cfg Config) *Authorizer {
 
 // Authorize will give full access (to any resource!) if the user is part of
 // the admin list or no access at all if they are not
-func (a *Authorizer) Authorize(principal *models.Principal, verb, resource string) error {
+func (a *Authorizer) Authorize(principal *models.Principal, verb string, resources ...string) error {
```
Comment (dirkkul) [inline usecases/auth/authorization/adminlist/authorizer.go:41]: can we have a custom type for verbs?

File: usecases/auth/authorization/types.go [hunk 2]
```diff
@@ -18,105 +18,140 @@ import (
...
+	// CRUD = "(C)|(R)|(U)|(D)"
+
+	// CREATE Represents the action to create a new resource.
+	CREATE = "C"
+	// READ Represents the action to retrieve a resource.
+	READ = "R"
+	// UPDATE Represents the action to update an existing resource.
+	UPDATE = "U"
+	// DELETE Represents the action to delete a resource.
+	DELETE = "D"
```
Comment (dirkkul) [inline usecases/auth/authorization/types.go:33]: this would be the custom type :)

File: usecases/objects/update.go [hunk 3]
```diff
@@ -30,8 +30,13 @@ func (m *Manager) UpdateObject(ctx context.Context, principal *models.Principal,
...
 ) (*models.Object, error) {
-	err := m.authorizer.Authorize(principal, authorization.UPDATE, authorization.Objects(class, id))
-	if err != nil {
+	// can read source
+	if err := m.authorizer.Authorize(principal, authorization.READ, authorization.Objects(class, "", id)); err != nil {
+		return nil, err
+	}
+
+	// can update target
+	if err := m.authorizer.Authorize(principal, authorization.UPDATE, authorization.Objects(updates.Class, updates.Tenant, updates.ID)); err != nil {
```
Comment (dirkkul) [inline usecases/objects/update.go:39]: Pretty sure the class + tenant has to be the same to update an object

File: usecases/auth/authorization/adminlist/authorizer.go (hunk 1 above)
Comment (moogacs) [reply usecases/auth/authorization/adminlist/authorizer.go:41]: that will require much bigger diff

File: usecases/auth/authorization/types.go (hunk 2 above)
Comment (moogacs) [reply usecases/auth/authorization/types.go:33]: that will require much bigger diff

File: usecases/auth/authorization/adminlist/authorizer.go (hunk 1 above)
Comment (moogacs) [reply usecases/auth/authorization/adminlist/authorizer.go:41]: but i agree

**Review by dirkkul — DISMISSED** [review]

**Review by dirkkul — APPROVED** [review]

**Pull request merged**